### ISSUE TITLE
[IntelliJ][ReportedByUser] Uncaught Exception  java.lang.NullPointerException

### ISSUE THREAD

**Issue by kyx42** [microsoft/azure-tools-for-java#3017]
IntelliJ build version: 2019.1 IU-191.6183.87
OS: Linux
JDK: JetBrains s.r.o 1.8.0_202-release
Plugin version: 3.20.0-2019.1
Additional Info: None
Parent component: 
```
com.intellij.openapi.wm.impl.IdeRootPane[,0,34,1865x1016,invalid,layout=com.intellij.openapi.wm.impl.IdeRootPane$MyRootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=449,maximumSize=,minimumSize=,preferredSize=]
```
Error message: 
```
IdeaLoggingEvent[message=null, throwable=java.lang.NullPointerException
	at com.microsoft.azure.management.resources.fluentcore.arm.models.implementation.ResourceImpl.id(ResourceImpl.java:71)
	at com.microsoft.azure.management.resources.fluentcore.arm.models.implementation.GroupableResourceImpl.resourceGroupName(GroupableResourceImpl.java:69)
	at com.microsoft.azure.management.compute.implementation.VirtualMachineImpl.refreshInstanceViewAsync(VirtualMachineImpl.java:384)
	at com.microsoft.azure.management.compute.implementation.VirtualMachineImpl.refreshInstanceView(VirtualMachineImpl.java:379)
	at com.microsoft.azure.management.compute.implementation.VirtualMachineImpl.instanceView(VirtualMachineImpl.java:1617)
	at com.microsoft.tooling.msservices.serviceexplorer.azure.vmarm.VMNode.isRunning(VMNode.java:241)
	at com.microsoft.tooling.msservices.serviceexplorer.azure.vmarm.VMNode.getNodeActions(VMNode.java:217)
	at com.microsoft.intellij.components.ServerExplorerToolWindowFactory.creat

**Comment by wezhang** [comment]
Thanks @kyx42 ! Hanxiao, could you help have a look?

**Comment by Flanker32** [comment]
@kyx42 Thanks for your report. could you please share the steps to repro this issue?

**Comment by Flanker32** [comment]
We didn't remove the tool window listener when close a project, fixed in #3431
